android/app/src/main/java/org/smartregister/fhircore/viewmodel/PatientListViewModel.kt

```diff
@@ -81,6 +96,14 @@ class PatientListViewModel(application: Application, private val fhirEngine: Fhi
           from = (page * pageSize)
         }
 
+      val showOnlyOverdue = SharedPrefrencesHelper.read(PatientListFragment.SHOW_OVERDUE_PATIENTS)
+      searchResults = searchResults.distinctBy { it.logicalId }
+      searchResults =
+        searchResults.filter {
+          !showOnlyOverdue or
+            (fetchPatientStatus(it.logicalId).value?.status == VaccineStatus.OVERDUE)
```

Comment:
`fetchPatientStatus` is a suspend function, meaning it may not return immediately. We should observer it's return value instead.

Also, I think we should only run this block if `showOnlyOverdue` is set to `true`.

Could we also rename this `showOnlyOverdue` to `shouldShowOnlyOverduePatients`.

Reply:
I think we can use value because `fetchPatientStatus` is already in async operation. I have checked it's working fine

Reply:
I have added if check and rename a variable